### PR TITLE
Connects auth with backend

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**

### CODE REVIEW

File: openslides_backend/services/authentication.py [hunk 1]
```diff
@@ -49,18 +48,42 @@ def get_user(self, headers: Headers) -> int:
...
+        """
+        return f"/{path}" if not path.startswith('/') else path
+
+    def get_internal_url(self, path: str) -> str:
+        """
+        Returns an internal url to a specified path at auth-service.
+        """
+        return f"{self.url}{AUTH_INTERNAL_URL_PREFIX}{self.format_url_path(path)}"
+
+    def get_external_url(self, path: str) -> str:
```
Comment (FinnStutzenstein) [inline openslides_backend/services/authentication.py:85]: not needed, can be deleted.

File: openslides_backend/services/authentication.py [hunk 2]
```diff
@@ -49,18 +48,42 @@ def get_user(self, headers: Headers) -> int:
...
+
             try:
-                user_id = body["user_id"]
+                user_id = body["userId"]
             except (TypeError, KeyError):
                 raise AuthenticationException(
                     "Empty or bad response from authentication service."
                 )
-        return user_id
+            try:
```
Comment (FinnStutzenstein) [inline openslides_backend/services/authentication.py:60]: `access_token = response.headers("Authentication", None)`

You can use `Optional[str]` then and `if access_token is not None:`.

File: openslides_backend/shared/interfaces.py [hunk 3]
```diff
@@ -47,7 +48,7 @@ class Services(Protocol):  # pragma: no cover
 
     # TODO: Use correct type here. Fitting together dependency_injector and our
```
Comment (FinnStutzenstein) [inline openslides_backend/shared/interfaces.py:49]: Can the todo be removed?

Comment (GabrielInTheWorld) [reply openslides_backend/shared/interfaces.py:49]: I guess, this is a todo for the other two lines under authentication (they're set as `any`, too).

**Comment by FinnStutzenstein** [comment]
Please rebase on the current master (fixes tests) and resolve all issues with mypy etc. You can use `make run-cleanup` for some auto-formatting and see the errors. If you have questions write me a message.

**Comment by FinnStutzenstein** [comment]
We will merge this for now, since it is still in development. Can be refined later.

**Pull request merged**